File: Lemonade/src/main/java/org/lemonade/expression/UnaryExpression.java
package org.lemonade.expression;

import org.lemonade.QLOperatorType;

public abstract class UnaryExpression extends Expression {
    private Expression expression;

    public UnaryExpression() {

    }

    public void setExpression(Expression expression) {
        this.expression = expression;
    }

    public Expression getExpression() {
        return expression;
    }

}

File: OffByOne/src/OffByOne.Ql/Ast/Expressions/Unary/Base/UnaryExpression.cs
namespace OffByOne.Ql.Ast.Expressions.Unary.Base
{
    using System;
    using System.Collections.Generic;

    public abstract class UnaryExpression : Expression
    {
        protected UnaryExpression(Expression expression)
        {
            if (expression == null)
            {
                throw new ArgumentNullException(nameof(expression));
            }

            this.Expression = expression;
        }

        public Expression Expression { get; }

        public override ISet<string> GetDependencies()
        {
            return this.Expression.GetDependencies();
        }
    }
}
